hooks/tk-multi-publish2/maya/publish_files_integ.py

```diff
@@ -0,0 +1,266 @@
+            :param:
+                group_nodes: the list of nodes in the scene
+        """
+        for name in range(len(group_nodes) - 1):
+            # Listing children of group nodes
+            children = cmds.listRelatives(group_nodes[name], c=True)
+            # group_nodes is arranged in hierarchial order i.e. the next node should be the child of previous
+            if children and (group_nodes[name + 1] in children):
+                if name == 'SET_TO_WORLD' and 'CAM' in children:
+                    continue
+            else:
+                hierarchy = "WORLDSCALE\n|__SET_TO_WORLD\n" + "    " + "|__TRACK_GEO\n" + "    " + "|__CAM"
+                self.logger.error("Incorrect hierarchy.",
+                                  extra={
+                                      "action_show_more_info": {
+                                          "label": "Show Info",
+                                          "tooltip": "Show the required hierarchy",
+                                          "text": "Required hierarchy:\n\n{}".format(hierarchy)
+                                      }
+                                  }
+                                  )
+                return False
+        return self._track_geo_child_naming(group_nodes[-1])
+
+    def _connected_image_plane(self, camera):
+        camshape = cmds.listRelatives(camera, s=True, c=True)[0]
+        connections = cmds.listConnections(camshape + '.imagePlane', source=True, type='imagePlane')
+        if not connections:
+            self.logger.error("Image plane not attached to CAM.")
+            return False
+        return True
+
+    def _camera_naming(self, group_nodes):
+        """Checks the naming of the camera.
+            :param:
+                group_nodes: The list of nodes that should be in the scene. This will be
+                used to check node hierarchy once camera naming is validated.
+        """
+        # Look for all the cameras present in the scene
+        all_cameras = cmds.listCameras()
+        # Remove the default_cameras from the list
+        main_cam = list(set(all_cameras) - set(_default_cameras))
+        if main_cam:
+            if len(main_cam) > 1:
+                # Checking if more than one CAM present
+                self.logger.error("More the one camera detected. Only CAM should be present.")
+                return False
+            elif main_cam[0] != _camera_name:
+                # Validating camera name
+                self.logger.error("Incorrect camera name! Should be CAM.")
+                return False
+        else:
+            self.logger.error("Camera (CAM) not present in the scene.")
+            return False
+        return self._connected_image_plane(main_cam[0])
+
+    def _node_naming(self, group_nodes, groups):
+        """Checking if the established group node names have not been tampered with.
+            :param:
+                group_nodes: group nodes to be present in the scene
```

Comment:
This is a global constant. Why are you passing it from method to method?

Reply:
Oh yes. Don't know how i did that. This should be group nodes in the scene not global constants.